document/CoreFormat.mdk

```diff
@@ -413,6 +413,99 @@ The `results` array shall be empty if the run that produced the `run` object did
 
+### `toolNotifications` { #run-toolNotifications }
+
+A `run` object may contain a property named `toolNotifications`
+whose value is an array of zero or more `notification` objects (&sect;[#notification]).
+Each element of the array represents a runtime condition detected by the tool.
+The presence within this array of any `notification` object whose `level` property (&sect;[#notification-level]) is `error` shall mean
+that the run failed.
+
+~ IsoNumberedNote { @IsoNumberedNote=1 }
+The information in `toolNotifications` is primarily intended for the developers of the analysis tool,
+to aid them in diagnosing bugs in the tool.
+This is in contrast to the information in `results`, which is intended for the developers of the code being analyzed.
+~
+
+~ IsoNumberedNote
+Depending on the nature of the error, a tool that encounters a runtime error might or might not be able to continue running.
+
+If the error occurs in the course of evaluating a rule, the tool might report the error in `toolNotifications`, disable the rule,
+and continue to execute the remaining rules.
+
+If the error occurs outside of the evaluation of a rule, the tool might report
+the error in `toolNotifications` and then halt.
+If the tool exits abnormally, it might not have the opportunity to report the error.
+~
+
+### `configurationNotifications` { #run-configurationNotifications }
+
+A `run` object may contain a property named `configurationNotifications`
+whose value is an array of zero or more `notification` objects (&sect;[#notification]).
+Each element of the array represents a condition relevant to the tool's configuration.
+The presence within this array of any `notification` object whose `level` property (&sect;[#notification-level]) is `error` shall mean
+that the run failed.
+
+~ IsoNumberedNote { @IsoNumberedNote=1 }
+The information in `toolNotifications` is primarily intended for the engineers who configure the analysis tool,
+to aid them in diagnosing errors in the configuration.
+This is in contrast to the information in `results`, which is intended for the developers of the code being analyzed.
+~
+
+~ IsoNumberedNote
+Many tools can be parameterized with information about which rules to run, and how they should be configured.
+In some cases, if the configuration information is invalid, the tool can ignore the invalid information
+and continue to run.
+~
+
+~ IsoNumberedExample { @IsoNumberedExample=1 }
+A tool is invoked with a configuration file which specifies that the tool should disable rule `ABC0001`,
+but there is no rule whose `id` is `ABC0001`.
+The tool should report the problem in `configurationNotifications`. The tool might
+continue to run, reporting results for the rules that are correctly configured.
+```
+"configurationNotifications": [
+    {
+        "id": "UnknownRule",
+        "ruleId": "ABC0001",
+        "level": "warning",
+        "message": "The configuration file specifies that an unknown rule \"ABC0001\" is to be disabled." 
```

Comment:
In practice as a tools developer, I would simpler, active voice language for this.
'Could not located specified rule 'ABC0001' in order to disable it.'